crates/iceberg-rest-server/src/catalog/tables.rs

```diff
@@ -779,3 +827,33 @@ fn validate_table_ident(table: &TableIdent) -> Result<()> {
     }
     Ok(())
 }
+
+// This function does not return a result but serde_json::Value::Null if serialization of tab.table
+// fails. This follows the rationale that we'll likely end up ignoring the error in the API handler
+// anyway since we already effected the change and only the event emission about the change failed.
+// Given that we are serializing stuff that we've received as path parameters and successfully
+// processed, it's unlikely to cause issues.
+fn maybe_table_parameters_to_json(tab: &TableParameters) -> serde_json::Value {
+    let table_json = if let Ok(table) = serde_json::to_value(&tab.table) {
+        table
+    } else {
+        tracing::warn!("Failed to convert table parameters to json, they will be missing from the published event.");
+        serde_json::Value::Null
+    };
+    serde_json::json!({"prefix": tab.prefix.as_ref().map(|pre| pre.as_str().to_string()),
+                           "table": table_json})
+}
+
+// This function does not return a result but serde_json::Value::Null if serialization of tab.table
+// fails. This follows the rationale that we'll likely end up ignoring the error in the API handler
+// anyway since we already effected the change and only the event emission about the change failed.
+// Given that we are serializing stuff we've received as a json body and also successfully
+// processed, it's unlikely to cause issues.
+fn maybe_body_to_json(request: impl Serialize) -> serde_json::Value {
```

Comment:
This function might even become obsolete if we follow my suggestion from above to split the body from metadata attributes, as error handling if serialization fails is already part of the publisher

Reply:
That'd mean we need to make all requests (and their fields) clone and we will end up doing Request -> cloned Request -> Json, I agree that we get clearer encapsulation